packages/@aws-cdk/aws-ecs/lib/load-balanced-service.ts

```diff
@@ -0,0 +1,133 @@
+import { ICertificate } from '@aws-cdk/aws-certificatemanager';
+import elbv2 = require('@aws-cdk/aws-elasticloadbalancingv2');
+import cdk = require('@aws-cdk/cdk');
+import { BaseService } from './base/base-service';
+import { ICluster } from './cluster';
+import { IContainerImage } from './container-image';
+
+export enum LoadBalancerType {
+  Application,
+  Network
+}
+
+export interface LoadBalancedServiceProps {
+  /**
+   * The cluster where your service will be deployed
+   */
+  cluster: ICluster;
+
+  /**
+   * The image to start.
+   */
+  image: IContainerImage;
+
+  /**
+   * The container port of the application load balancer attached to your Fargate service. Corresponds to container port mapping.
+   *
+   * @default 80
+   */
+  containerPort?: number;
+
+  /**
+   * Determines whether the Application Load Balancer will be internet-facing
+   *
+   * @default true
+   */
+  publicLoadBalancer?: boolean;
+
+  /**
+   * Number of desired copies of running tasks
+   *
+   * @default 1
+   */
+  desiredCount?: number;
+
+  /**
+   * Whether to create an application load balancer or a network load balancer
+   * @default application
+   */
+  loadBalancerType?: LoadBalancerType
+
+  /**
+   * Environment variables to pass to the container created for the service
+   * @default No environment variables
+   */
+  containerEnvironment?: {[key: string]: string};
+
+  /**
+   * Certificate Manager certificate to associate with the load balancer.
+   * Setting this option will set the load balancer port to 443.
+   */
+  certificate?: ICertificate;
+}
+
+/**
+ * Base class for load-balanced Fargate and ECS service
+ */
+export abstract class LoadBalancedService extends cdk.Construct {
+  public readonly loadBalancerType: LoadBalancerType;
+
+  public readonly loadBalancer: elbv2.BaseLoadBalancer;
+
+  public readonly listener: elbv2.ApplicationListener | elbv2.NetworkListener;
+
+  public readonly targetGroup: elbv2.ApplicationTargetGroup | elbv2.NetworkTargetGroup;
+
+  constructor(scope: cdk.Construct, id: string, props: LoadBalancedServiceProps) {
+    super(scope, id);
+
+    // Load balancer
+    this.loadBalancerType = props.loadBalancerType !== undefined ? props.loadBalancerType : LoadBalancerType.Application;
+
+    if (this.loadBalancerType !== LoadBalancerType.Application && this.loadBalancerType !== LoadBalancerType.Network) {
+       throw new Error(`invalid loadBalancerType`);
+    }
+
+    const internetFacing = props.publicLoadBalancer !== undefined ? props.publicLoadBalancer : true;
+
+    const lbProps = {
+      vpc: props.cluster.vpc,
+      internetFacing
+    };
+
+    if (this.loadBalancerType === LoadBalancerType.Application) {
+      this.loadBalancer = new elbv2.ApplicationLoadBalancer(this, 'LB', lbProps);
+    } else {
+      this.loadBalancer = new elbv2.NetworkLoadBalancer(this, 'LB', lbProps);
+    }
+
+    const targetProps = {
+      port: 80
```

Comment:
Shouldn't this use `containerPort`?

Reply:
I thought so too, but apparently [when used with ECS the port number for the target group is ignored](https://stackoverflow.com/questions/42715647/whats-the-target-group-port-for-when-using-application-load-balancer-ec2-con).

I can confirm that it works fine as-is, I use 8000 for my container port.